packages/stores/src/providers/WebsiteStatusProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useSubscription } from '@deriv/api';
+import useStore from '../useStore';
+import { useWS } from '@deriv/shared';
+import merge from 'lodash.merge';
+
+const WebsiteStatusProvider = ({ children }: React.PropsWithChildren<unknown>) => {
+    const { data, is_subscribed, subscribe, unsubscribe } = useSubscription('website_status');
+    const {
+        client: { is_logged_in, is_logging_in, loginid },
+        website_status: { update },
+    } = useStore();
+    const WS = useWS();
+
+    React.useEffect(() => {
+        if (data) update(prev => merge(prev, data));
+    }, [update, data]);
+
+    React.useEffect(() => {
+        // handles case where the response from website_status is different when client is logged in/out, we need to re-subscribe in that case
+        // refer to socket-general.js
+        if (is_subscribed) unsubscribe();
+        if (is_logged_in || is_logging_in) {
+            WS.get()
+                .expectResponse('authorize')
+                .then(() => {
+                    subscribe();
+                });
+        } else {
+            subscribe();
```

Comment:
@adrienne-deriv I don't think we need to make it complicated, We just need to resubscribe whenever a new authorization happens 🤔 

```tsx
import React, { useEffect } from 'react';
import { useSubscription } from '@deriv/api';
import { observer } from 'mobx-react-lite';
import useStore from '../useStore';

const WebsiteStatusProvider = observer(({ children }: React.PropsWithChildren<unknown>) => {
    const { is_subscribed, data, subscribe, unsubscribe } = useSubscription('website_status');
    const {
        client: { is_authorize },
        website_status: { update },
    } = useStore();

    useEffect(() => {
        if (data) update(data);
    }, [update, data]);

    useEffect(() => {
        if (!is_authorize) unsubscribe();
        if (!is_subscribed && is_authorize) subscribe();
    }, [is_authorize, is_subscribed, subscribe, unsubscribe]);

    return <>{children}</>;
});

export default WebsiteStatusProvider;
```

Would you please help to check the above code 🙏🏻 Please note that the above code won't subscribe if the user is not logged in, I'm not sure if we need to do that or not but if we need then we can update the condition for subscribe 🤔

Reply:
@adrienne-deriv please correct me if I'm wrong, but based on [this](https://github.com/binary-com/deriv-app/pull/8702/files#r1203826550) comment, we should subscribe in both cases whether the client is logged in or not.  
if the only difference in the data is `p2p_config` , can't we just update the data and remove p2p_config instead of re-subscribing?

Reply:
The only time we need to re-subscribe is when the user is logged out, and they start logging in and authenticating. When user is logging out, we can continue subscribing